R/Filtering.R

```diff
@@ -240,3 +240,161 @@ TADA_FieldValuesTable <- function(.data, field = "null", characteristicName = "n
+                                      surface_water = TRUE,
+                                      ground_water = FALSE,
+                                      sediment = FALSE) {
+  
+  # import MonitoringLocationTypeNames and TADA.Groundwater.Flags
+  sw.sitetypes <- utils::read.csv(system.file("extdata", "WQXMonitoringLocationTypeNameRef.csv", package = "TADA")) %>%
+    dplyr::select(MonitoringLocationTypeName, TADA.Groundwater.Flag) %>%
+    dplyr::rename(ML.Groundwater.Flag = TADA.Groundwater.Flag)
+  
+  
+  # add TADA.Groundwater.Flag column
+  .data <- .data %>%
+    # identify TADA.Groundwater.Flag using ActivityMediaSubdivisionName and columns related to groundwater
+    dplyr::mutate(TADA.Groundwater.Flag = dplyr::case_when(ActivityMediaSubdivisionName == "Groundwater" ~ "Groundwater",
+                                                           !is.na(AquiferName) |
+                                                             !is.na(AquiferTypeName) |
+                                                             !is.na(LocalAqfrName) |
+                                                             !is.na(ConstructionDateText) |
+                                                             !is.na(WellDepthMeasure.MeasureValue) |
+                                                             !is.na(WellDepthMeasure.MeasureUnitCode) |
+                                                             !is.na(WellHoleDepthMeasure.MeasureValue) |
+                                                             !is.na(WellHoleDepthMeasure.MeasureUnitCode) ~ "Groundwater",
+                                                           ActivityMediaSubdivisionName == "Surface Water" ~ "Surface Water",
+                                                           TADA.ActivityMediaName != "WATER" ~ "Not Water")) %>%
+    # add TADA.Groundwater.Flag for additional rows based on MonitoringLocationTypeName
+    dplyr::left_join(sw.sitetypes) %>%
+    dplyr::mutate(TADA.Groundwater.Flag = ifelse(is.na(TADA.Groundwater.Flag),
+                                                 ML.Groundwater.Flag, TADA.Groundwater.Flag)) %>%
+    dplyr::select(-ML.Groundwater.Flag)
+  
+  { if (surface_water == TRUE)
+    
+    sur.water.data <- .data %>%
+      dplyr::filter(TADA.Groundwater.Flag == "Surface Water") %>%
+      dplyr::mutate(TADA.UseForAssessment.Flag = "Yes")
+    }
+  
+  { if (surface_water == FALSE)
+    
+    sur.water.data <- .data %>%
+      dplyr::filter(TADA.Groundwater.Flag == "Surface Water") %>%
+      dplyr::mutate(TADA.UseForAssessment.Flag = "No")
+    
+  }
+  
+  
+  if (ground_water == TRUE) {
+    
+    gr.water.data <- .data %>%
+      dplyr::filter(TADA.Groundwater.Flag == "Groundwater") %>%
+      dplyr::mutate(TADA.UseForAssessment.Flag = "Yes")
+    
+  }
+  
+  if (ground_water == FALSE) {
+    gr.water.data <- .data %>%
+      dplyr::filter(TADA.Groundwater.Flag == "Groundwater") %>%
+      dplyr::mutate(TADA.UseForAssessment.Flag = "No")
```

Comment:
Instead of creating separate TADA.Groundwater.Flag and sediment flag columns, can we include this information in the TADA.UseForAssessment.Flag?

Values in the TADA.UseForAssessment.Flag could be more specific than Yes/No to consolidate these extra columns. For example, if the row is GW, then the TADA.UseForAssessment.Flag could be "Groundwater" or "No-Groundwater". 

Same for sediment and surface water ("No-Sediment" or "Yes-Surfacewater".)